### ISSUE TITLE
[Bug]: refresh_continuous_aggregate doesn't refresh hierarchical caggs

### ISSUE THREAD

**Issue by leppaott** [timescale/timescaledb#6306]
### What type of bug is this?

Other

### What subsystems and features are affected?

Other

### What happened?

Hey, noticed on CI e2e tests that when we have 5min, 15min, hourly etc caggs on top of each like:

```
CREATE MATERIALIZED VIEW "HistoryAgg5Minute"
WITH (timescaledb.continuous) AS
SELECT time_bucket(INTERVAL '5 minutes', "time") AS "bucket",
  uuid,
  state_agg("time", "status") AS "statusAgg"
  FROM "StatusHistory"
  GROUP BY 1, 2
WITH NO DATA;
```

Noticed the **Bug 1 Inserts into the bottom hypertable aren't real-time reflected on hierarchical aggs.** Results are valid only for the lowest hierarchical agregation - the 5min in this case.

Then I wrote a workaround to force it with `refresh_continuous_aggregate` and noticed the **Bug 2 calling refresh on the highest cagg doesn't refresh any of the bottom hierarchical aggregations** not even itself with the correct results.

So I end up writing code like:

```
for (const aggregate of ['5Min', '15min', 'largestLastAgg') {
      await execute(
        `CALL refresh_continuous_aggregate('"${aggregate}"', null, null);`
      );
```

to refresh from the bottom aggregate to the top.

I think not refreshing the whole aggregate hierarchy doesn't make too much sense or there should be a flag on refresh_continuous_aggregate.

### TimescaleDB version affected

2.12.2

### PostgreSQL version used

15.4

### What operating system did you use?

Debian

### What installation method did you use?

Docker

### What platform did you run on?

Other

### How can we reproduce the bug?

```bash
Have hierarchy of caggs, on tests insert into the hypertable and immediately query the results on the top aggregation. If it matter `state_agg` was used and `rollup()` on top layers.
```

**Comment by leppaott** [comment]
```
2023-11-10 08:59:17.262 UTC [753] LOG:  refreshing continuous aggregate "HistoryAgg5Minute" in window [ 2023-11-07 09:00:00+00, 2023-11-10 08:50:00+00 ]
2023-11-10 08:59:17.843 UTC [750] WARNING:  compressing chunk "_timescaledb_internal._hyper_1_16_chunk" failed when compression policy is executed
2023-11-10 08:59:17.843 UTC [750] DETAIL:  Message: (deadlock detected), Detail: (Process 750 waits for AccessShareLock on relation 35813 of database 34904; blocked by process 720.
	Process 720 waits for ShareLock on transaction 3692; blocked by process 750.).
2023-11-10 08:59:17.843 UTC [750] CONTEXT:  PL/pgSQL function _timescaledb_functions.policy_compression_execute(integer,integer,anyelement,integer,boolean,boolean) line 56 at RAISE
	SQL statement "CALL _timescaledb_functions.policy_compression_execute(
	        job_id, htid, lag_value::INTERVAL,
	        maxchunks, verbose_log, recompress_enabled
	      )"
	PL/pgSQL function _timescaledb_functions.policy_compression(integer,jsonb) line 51 at CALL
```
This compression policy might have something to do with this as well..

**Comment by mahipv** [comment]
Hi @leppaott, Thank you for reporting the issue.
The lower caggs are not automatically refreshed while refreshing stacked caggs. There is a feature request #6010 to chain the refresh of the hierarchical caggs. Would that solve the issue here?

**Comment by leppaott** [comment]
@mahipv would that help with real-time refreshing on the stacked caggs as well? Sounds like it would help if there would be a way to refresh a stacked aggregate automatically after the base cagg is refreshed. So in that request after every 2 minutes also the 10 minute caggs. But of course assuming it would be still possible for stacked aggs to be refreshed only every 10 minutes (in this case can it ensure it's done after base agg? Would think if the policies are in order).

**Comment by mahipv** [comment]
Marking this as a duplicate of #6010. Please feel to reopen the issue if you think otherwise.
